Workspaces/MiniWorldWorkspace/MiniWorldWorkspace.cs

```diff
@@ -219,42 +207,50 @@ void Update()
 			m_GridMaterial.SetVector("_ClipCenter", inverseRotation * m_MiniWorld.transform.position);
 		}
 
-		public void ProcessInput(ActionMapInput input, ConsumeControlDelegate consumeControl)
+		public override void ProcessInput(ActionMapInput input, ConsumeControlDelegate consumeControl)
 		{
-			var miniWorldInput = (MiniWorldInput)input;
+			base.ProcessInput(input, consumeControl);
+			var workspaceInput = (WorkspaceInput)input;
 
-			if (miniWorld.Contains(leftRayOrigin.position) && miniWorldInput.leftGrab.wasJustPressed)
+			var leftGrip = workspaceInput.secondaryLeft;
+			if (leftGrip.wasJustPressed && miniWorld.Contains(leftRayOrigin.position))
 			{
 				OnPanZoomDragStarted(leftRayOrigin);
-				consumeControl(miniWorldInput.leftGrab);
+				consumeControl(leftGrip);
 			}
 
-			if (miniWorld.Contains(rightRayOrigin.position) && miniWorldInput.rightGrab.wasJustPressed)
+			var rightGrip = workspaceInput.secondaryRight;
+			if (rightGrip.wasJustPressed && miniWorld.Contains(rightRayOrigin.position))
 			{
 				OnPanZoomDragStarted(rightRayOrigin);
-				consumeControl(miniWorldInput.rightGrab);
+				consumeControl(rightGrip);
 			}
 
-			if (miniWorldInput.leftGrab.isHeld || miniWorldInput.rightGrab.isHeld)
+			if (leftGrip.isHeld || rightGrip.isHeld)
```

Comment:
There's no reason not to...

I like keeping the logic separated, because that way we don't have to call OnPanZoomDragging in two places, and we semantically separate DragStarted, Dragging, and DragEnded in distinct conditions.